astropy/coordinates/sky_coordinate.py

```diff
@@ -41,6 +42,37 @@ def FRAME_ATTR_NAMES_SET():
     return out
 
 
+class SkyCoordDataInfo(BaseInfo):
+    attrs_from_parent = set(['unit'])
+
+    @staticmethod
+    def default_format(val):
```

Comment:
My original idea here was to make this a property, which returns a format string that will be parsed by `pprint`, i.e., something like the following:
```
@property
def default_format(self):
    components = self._repr_data.components
    formats = ['{0.' + compname + '.value:}' for compname in components]
    return ','.join(formats)
```
I think this might not work since you seem to change the representation in `_repr_data`, but I don't understand `SkyCoord` well enough to see why this is needed; it seems you do something special mostly for sperical without a distance, but in that case I'd think one could do with just not including the distance among the components.

Anyway, mostly in case this can be done more simply;